plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/use-form-fields.ts

```diff
@@ -62,6 +81,7 @@ export const useFormFields = (
 			if (
 				defaultConfig.rules.hidden &&
 				typeof defaultConfig.rules.hidden === 'object' &&
+				! Array.isArray( defaultConfig.rules.hidden ) &&
 				Object.keys( defaultConfig.rules.hidden ).length > 0
```

Comment:
Whats all this then? Same question for required above. Why are all these conditionals necessary? If this is necessary, could you make a helper that works for all rule types?

Reply:
The extra change was needed to account for an empty array that comes from the server when no rules are set, I can simplify this set of rules thou.